FWCore/Framework/src/limited/EDProducerBase.cc

```diff
@@ -75,6 +76,33 @@ namespace edm {
 
     void EDProducerBase::doEndJob() { this->endJob(); }
 
+    void EDProducerBase::doBeginProcessBlock(ProcessBlockPrincipal const& pbp, ModuleCallingContext const* mcc) {
+      ProcessBlock processBlock(pbp, moduleDescription_, mcc, false);
+      processBlock.setConsumer(this);
+      ProcessBlock const& constProcessBlock = processBlock;
+      this->doBeginProcessBlock_(constProcessBlock);
+      processBlock.setProducer(this);
+      this->doBeginProcessBlockProduce_(processBlock);
+      commit_(processBlock);
+    }
+
+    void EDProducerBase::doAccessInputProcessBlock(ProcessBlockPrincipal const& pbp, ModuleCallingContext const* mcc) {
+      ProcessBlock processBlock(pbp, moduleDescription_, mcc, false);
+      processBlock.setConsumer(this);
```

Comment:
Just an observation. It really seems like the Principal wrappers should have their constructors changed so that the consumer is passed to the constructor. Calling a constructor followed immediately by a `set` function smells bad.

Reply:
From the way you worded the comment, I am assuming this is something we'll deal with in a separate PR later. This pattern is simply copied from what we already do for Runs, Lumis, and Events. It is not new with ProcessBlock.

Reply:
Let's look into it in a separate PR, I opened an issue to remind us https://github.com/cms-sw/cmssw/issues/30317.